synapse/app/homeserver.py

```diff
@@ -348,6 +348,23 @@ def setup(config_options: List[str]) -> SynapseHomeServer:
     if config.server.gc_seconds:
         synapse.metrics.MIN_TIME_BETWEEN_GCS = config.server.gc_seconds
 
+    if (
+        config.registration.enable_registration
+        and not config.registration.enable_registration_without_verification
+    ):
+        if (
+            not config.captcha.enable_registration_captcha
+            and not config.registration.registrations_require_3pid
+            and not config.registration.registration_requires_token
+        ):
+
+            raise ConfigError(
+                "You have enabled open registration without any verification. This is a known vector for "
+                "spam and abuse. If you have enabled registration, please add email, capcha, or token-based verification, or "
+                "use the config option `enable_registration_without_verification` in conjunction with `enable_verification."
```

Comment:
```suggestion
                "spam and abuse. If you would like to allow public registrations, please consider adding email, captcha, or token-based verification. "
                "Otherwise this check can be removed by setting the `enable_registration_without_verification` config option to `true`."
```

may be a bit clearer.